### PR TITLE
feat(linter): eslint/guard-for-in

### PR DESCRIPTION
**Pull request by JoSeBu1**
Rule details: https://eslint.org/docs/latest/rules/guard-for-in

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/guard_for_in.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                                return;
+                            }
+                        }
+                    }
+                }
+                _ => {}
+            }
+            ctx.diagnostic(GuardForInDiagnostic(Span::new(
+                for_in_statement.span.start,
+                for_in_statement.span.end,
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/eslint/guard_for_in.rs:71]: Well done, just one nitpick: maybe the `for..in` loop has a long body, is this possible to make the span position smaller?

Comment (mysteryven) [reply crates/oxc_linter/src/rules/eslint/guard_for_in.rs:71]: Do you think the first statement of `for..in`‘s body  is suitable?

Comment (JoSeBu1) [reply crates/oxc_linter/src/rules/eslint/guard_for_in.rs:71]: I don't know what can be done, I will take a look at it reviewing another snapshots

Comment (mysteryven) [reply crates/oxc_linter/src/rules/eslint/guard_for_in.rs:71]: Just advice, It's also good now!

**Review by mysteryven — APPROVED** [review]
Thank you!

**Pull request merged**